src/APICore.ts

```diff
@@ -211,7 +213,8 @@ export default class API {
     }
 
     private async request<T>(route: string, init: RequestInit, responseBodyFormat?: ResponseBodyFormat): Promise<T> {
-        const fetcher = createFetcher(this.getUrlFromRoute(route), init, isTooManyRequests);
+        const {url: enrichedRoute, ...enrichedRequestInit} = handleRequest(route, init, this.config.requestHandlers);
```

Comment:
Here, I pass the route of the request to the method that handles requests (`handleRequest`) to allow the possibility of modifying it if necessary.